### ISSUE TITLE
Prevent update of container if hook fails

### ISSUE THREAD

**Issue by nelsongraca** [containrrr/watchtower#1756]
### Is your feature request related to a problem? Please describe.

[The docs](https://containrrr.dev/watchtower/lifecycle-hooks/#execution_failure) say the following:

> The failure of a command to execute, identified by an exit code different than 0 or 75 (EX_TEMPFAIL), will not prevent watchtower from updating the container. Only an error log statement containing the exit code will be reported.

From this I understand that there is no way to prevent an update if a hook fails.

Reading this issue: https://github.com/containrrr/watchtower/issues/499 and the code in: https://github.com/containrrr/watchtower/blob/36391b0ae791ef9a451fd63b34b117689f6248eb/internal/actions/update.go#L158-167

Seems to say otherwise, if the pre-update hook has an exit code different than 0 it will stop and not update the container, the opposite of the docs.

Which is correct?

### Describe the solution you'd like

Make code and docs consistent or more clear

### Describe alternatives you've considered

none here

**Comment by piksel** [comment]
Yeah, this is really confusing. The docs are either just not updated, or are just referring to another lifecycle hook...
Either way, it doesn't make sense right now and does not reflect the actual behavior.

**Comment by piksel** [comment]
Yeah, it applies to all the lifecycle hooks _except_ `PreUpdate`. This should be clarified.

**Comment by nelsongraca** [comment]
@piksel Thanks for clearing it up, if there's something worse than no documentation it's wrong documentation
